test/gurka/test_matrix_time_dependent.cc

```diff
@@ -396,3 +398,123 @@ TEST(StandAlone, CostMatrixDeadends) {
     check_matrix(res_doc, {0.8f}, false, Matrix::CostMatrix);
   }
 }
+
+TEST(StandAlone, CostMatrixShapes) {
+  // keep the same order in the map.nodes for encoding easily
+  const std::string ascii_map = R"(
+    A-B-C-D-E-F-G-H-I-J-K-------L
```

Comment:
it's 3 ways but in order to keep the order in the `map.nodes`  `std::map` I made them contiguous, making it harder to see what's happening